sdk/storage/Azure.Storage.Blobs/tests/ProgressHandlingTests.cs

```diff
@@ -0,0 +1,178 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Azure.Core.TestFramework;
+using Azure.Storage.Blobs.Models;
+using Azure.Storage.Test;
+using Azure.Storage.Test.Shared;
+using NUnit.Framework;
+
+namespace Azure.Storage.Blobs.Tests
+{
+    public class ProgressHandlingTests : BlobTestBase
+    {
+        public ProgressHandlingTests(bool async, BlobClientOptions.ServiceVersion serviceVersion)
+            : base(async, serviceVersion, RecordedTestMode.Record /* RecordedTestMode.Record /* to re-record */)
+        {
+        }
+
+        [RecordedTest]
+        [TestCase(Constants.KB)] // simple case
+        [TestCase(256 * Constants.KB)] // multi-report case
```

Comment:
Did we record 256 KB of data here?  Might not matter but seems like a lot.

Reply:
Without a way to control progress reporting increments like we can with storage transfer options, this number isn't too big while guaranteeing beyond shadow of a doubt we hit incrementing reports in a test.